app/equip/potion.py

```diff
@@ -0,0 +1,12 @@
+class Potion:
+    def __init__(self, potion: dict):
+        self.potion = potion
+
+    def get_potion(self, hero) -> None:
+        if self.potion is not None:
+            if "hp" in self.potion["effect"]:

```

Comment:
These code duplications can be fixed, using `stats = ["protection", "power", "hp"]` and looping on `stats`